cwltool/main.py

```diff
@@ -1172,12 +1174,25 @@ def main(
                 print(f"{args.workflow} is valid CWL.", file=stdout)
                 return 0
 
-            if args.print_rdf:
+            if args.print_rdf or args.provenance:
+                output = stdout
+                if args.provenance:
+                    # No output variable set... and maybe no args.rdf_serializer?
+                    workflow_provenance = args.provenance + "/workflow.ttl"
+                    # Sets up a turtle file for the workflow information (not yet in the provenance folder as it does
+                    # not exist and creating it will give issues).
+                    output = open("workflow.ttl", "w")
```

Comment:
Better to save to a temporary folder instead of the current working directory

Reply:
Temp directory added in latest commit and removal of the temp directory